MANUAL.md

```diff
@@ -1600,158 +1692,748 @@ if (field.fieldKind == "map") {
 
 ## Writing plugins
 
-> **TODO** see v1 docs https://github.com/bufbuild/protobuf-es/blob/v1.10.0/docs/writing_plugins.md, and the plugin example in packages/protoplugin-example
+Code generator plugins are a unique feature of Protobuf compilers like `protoc` and the [Buf CLI][buf-cli]. With a
+plugin, you can generate files based on Protobuf schemas as the input. They can generate outputs like RPC clients and
+server stubs, mappings from Protobuf to SQL, validation code, and pretty much anything else you can think of can all be produced.
 
-## Examples
+The contract between the Protobuf compiler and a code generator plugin is defined in [plugin.proto]. Plugins are simple
+executables (typically on your `$PATH`) named `protoc-gen-x`, where `x` is the name of the language or feature that the
+plugin provides. The Protobuf compiler parses the Protobuf files and invokes the plugin, sending a
+`CodeGeneratorRequest` on stdin, and expecting a `CodeGeneratorResponse` on stdout. The request contains a set of
+descriptors—an abstract version of the parsed Protobuf files (see [descriptor.proto]). The response contains a list of
+files, each with a name and text content.
 
-> **TODO** packages/protobuf-example contains a runnable example that uses the same example.User definition we've been using in this doc. We should link to it.
+For more information on how plugins work, check out [the Buf image documentation][buf-images].
+TODO: Is this still the best place to link to? It's pretty Buf-specific in this more general context.
```

Comment:
The page has a section "How Protobuf plugins work" that fits, but I don't think it's worth still linking to it.

```suggestion
```